overlord/devicestate/devicectx.go

```diff
@@ -104,6 +104,18 @@ func (dc groundDeviceContext) HasModeenv() bool {
 	return dc.model.Grade() != asserts.ModelGradeUnset
 }
 
+// IsCoreBoot is true when there are modes, or when there are not but
+// we are not in classic (UC16/18 case)
+func (d *groundDeviceContext) IsCoreBoot() bool {
+	isCoreLegacy := !d.HasModeenv() && !d.Classic()
+	return d.HasModeenv() || isCoreLegacy
```

Comment:
isn't this simply  d.HasModeenv() || !d.Classic() ?

Reply:
Very true, changed now